src/module/actor/SoakFlow.ts

```diff
@@ -0,0 +1,102 @@
+import {ShadowrunRoll, ShadowrunRoller} from '../rolls/ShadowrunRoller';
+import {SR5Actor} from "./SR5Actor";
+import {SoakRules} from "./SoakRules";
+import { Helpers } from '../helpers';
+import SoakRollOptions = Shadowrun.SoakRollOptions;
+import ModList = Shadowrun.ModList;
+import {createRollChatMessage} from "../chat";
+import DamageData = Shadowrun.DamageData;
+import DamageElement = Shadowrun.DamageElement;
+import {PartsList} from "../parts/PartsList";
+import {DefaultValues} from "../dataTemplates";
+import { ShadowrunActorDialogs } from '../apps/dialogs/ShadowrunActorDialogs';
+
+export class SoakFlow {
+
+    /**
+     * Runs the soak flow with user interaction
+     * @param actor The actor doing the soaking
+     * @param soakRollOptions Information about the incoming damage (if it is already known)
+     * @param partsProps Optional modifiers for the soak test
+     */
+    async run(actor: SR5Actor, soakRollOptions: SoakRollOptions, partsProps: ModList<number> = []): Promise<ShadowrunRoll|undefined> {
+        // Figure out soak mods and damage
+        const soakDefenseParts = new PartsList<number>(partsProps);
+        SoakRules.applyDamageIndependentSoakParts(soakDefenseParts, actor);
+
+        const damageDataOrUndef = await this.promptDamageData(soakRollOptions, soakDefenseParts); 
+        if (damageDataOrUndef === undefined) {
```

Comment:
Nitpick: Ist expliziter Typencheck notwendig oder genügt !<expression>

Reply:
Will add it this in the next PR